### PR TITLE
Add ZhipuAI model wrappers

### PR DESCRIPTION
**Pull request by garyzhang99**
---
name: Add ZhipuAI model wrappers
about: Create a pull request
---
## Description

Support ZhipuAI API in agentscope.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: src/agentscope/models/zhipu_model.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        # TODO: set quota to the following metrics
+        self.monitor.register(
+            self._metric("call_counter"),
+            metric_unit="times",
+        )
+        self.monitor.register(
+            self._metric("prompt_tokens"),
+            metric_unit="token",
+        )
+        self.monitor.register(
```
Comment (ZiTao-Li) [inline src/agentscope/models/zhipu_model.py:341]: does embedding model response have "completion_tokens" information?

**Review by ZiTao-Li — COMMENTED** [review]
see inline comments

File: docs/sphinx_doc/en/source/tutorial/203-model.md [hunk 2]
```diff
@@ -302,6 +305,48 @@ Here we provide example configurations for different model wrappers.
...
+```python
+{
+    "config_name": "my_zhipuai_embedding_config",
+    "model_type": "zhipuai_embedding",
+
+    # Required parameters
+    "model_name": "{model_name}",               # The model name in ZhipuAI API, e.g. embedding-2
+
+    # Optional parameters
+    "api_key": "{your_api_key}",                #
```
Comment (DavdGao) [inline docs/sphinx_doc/en/source/tutorial/203-model.md:341]: remove #

File: docs/sphinx_doc/zh_CN/source/tutorial/203-model.md [hunk 3]
```diff
@@ -322,6 +325,48 @@ API如下：
...
+```python
+{
+    "config_name": "my_zhipuai_embedding_config",
+    "model_type": "zhipuai_embedding",
+
+    # Required parameters
+    "model_name": "{model_name}",               # The model name in ZhipuAI API, e.g. embedding-2
+
+    # Optional parameters
+    "api_key": "{your_api_key}",                #
```
Comment (DavdGao) [inline docs/sphinx_doc/zh_CN/source/tutorial/203-model.md:361]: remove #

File: src/agentscope/models/zhipu_model.py [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+        if zhipuai is None:
+            raise ImportError(
+                "Cannot find zhipuai package in current python environment.",
+            )
+
+        self.model_name = model_name
+        self.generate_args = generate_args or {}
+
+        self.client = zhipuai.ZhipuAI(
```
Comment (DavdGao) [inline src/agentscope/models/zhipu_model.py:58]: Do we need to consider other arguments here to initialize a client for ZhipuAI API?

**Review by DavdGao — COMMENTED** [review]
plz see inline comments

File: src/agentscope/models/zhipu_model.py (hunk 4 above)
Comment (garyzhang99) [reply src/agentscope/models/zhipu_model.py:58]: Added **kwargs to clients.

Comment (garyzhang99) [reply src/agentscope/models/zhipu_model.py:58]: In the ZhipuAi's examples, they only use the api_key, but actually arguments such as `timeout` can be set.

File: src/agentscope/models/zhipu_model.py (hunk 1 above)
Comment (garyzhang99) [reply src/agentscope/models/zhipu_model.py:336]: They include a `"completion_tokens":0` in their examples, so I added this just to be save. (See https://open.bigmodel.cn/dev/api#text_embedding)

File: src/agentscope/models/zhipu_model.py [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+            },
+            response=response.model_dump(),
+        )
+
+        # step4: update monitor accordingly
+        self.update_monitor(call_counter=1, **response.usage.model_dump())
+
+        # step5: return response
+        response_json = response.model_dump()
+        if len(response_json["data"]) == 0:
```
[the remark this answers is not in the document]
Comment (garyzhang99) [reply src/agentscope/models/zhipu_model.py]: Done, I was referencing the condition from OpenAI api, but for this we can just remove it.

File: src/agentscope/models/zhipu_model.py [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+            raise ImportError(
+                "Cannot find zhipuai package in current python environment.",
+            )
+
+        self.model_name = model_name
+        self.generate_args = generate_args or {}
+
+        self.client = zhipuai.ZhipuAI(
+            api_key=api_key,
+            **kwargs,
```
Comment (DavdGao) [inline src/agentscope/models/zhipu_model.py:60]: `client_args` is suggested here rather than `kwargs`

File: src/agentscope/models/zhipu_model.py [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+        Args:
+            config_name (`str`):
+                The name of the model config.
+            model_name (`str`, default `None`):
+                The name of the model to use in ZhipuAI API.
+            api_key (`str`, default `None`):
+                The API key for ZhipuAI API. If not specified, it will
+                be read from the environment variable.
+            client_args (`dict`, default `None`):
+                The extra keyword arguments to initialize the OpenAI client.
```
Comment (DavdGao) [inline src/agentscope/models/zhipu_model.py:45]: ZhipuAI, not OpenAI client

**Review by DavdGao — APPROVED** [review]
LGTM

**Pull request merged**